src/Interpreters/InterpreterCreateQuery.cpp

```diff
@@ -1880,7 +1880,7 @@ void InterpreterCreateQuery::addColumnsDescriptionToCreateQueryIfNecessary(ASTCr
     }
 }
 
-void InterpreterCreateQuery::processSQLSecurityOption(ContextPtr context_, ASTSQLSecurity & sql_security, bool is_attach, bool is_materialized_view)
+void InterpreterCreateQuery::processSQLSecurityOption(ContextPtr context_, ASTSQLSecurity & sql_security, bool is_attach, bool is_materialized_view, bool is_restore_from_backup_)
```

Comment:
`bool is_attach` and `bool is_restore_from_backup_` are processed in this function in exactly the same way. Let's join those parameters together into something like `bool check_permissions`.